crates/router/src/connector/cybersource.rs

```diff
@@ -59,25 +74,101 @@ impl Cybersource {
         let signature_value =
             consts::BASE64_ENGINE.encode(hmac::sign(&key, signature_string.as_bytes()).as_ref());
         let signature_header = format!(
-            r#"keyid="{api_key}", algorithm="HmacSHA256", headers="{headers_for_post_method}", signature="{signature_value}""#
+            r#"keyid="{api_key}", algorithm="HmacSHA256", headers="{headers}", signature="{signature_value}""#
         );
 
         Ok(signature_header)
     }
 }
 
-impl api::ConnectorCommon for Cybersource {
+impl ConnectorCommon for Cybersource {
     fn id(&self) -> &'static str {
         "cybersource"
     }
 
     fn common_get_content_type(&self) -> &'static str {
-        "application/json"
+        "application/json;charset=utf-8"
     }
 
     fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
         connectors.cybersource.base_url.as_ref()
     }
+
+    fn build_error_response(
+        &self,
+        res: Bytes,
+    ) -> CustomResult<types::ErrorResponse, errors::ConnectorError> {
+        let response: cybersource::ErrorResponse = res
+            .parse_struct("Cybersource ErrorResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        Ok(types::ErrorResponse {
+            code: consts::NO_ERROR_CODE.to_string(),
+            message: response
+                .message
+                .unwrap_or_else(|| consts::NO_ERROR_MESSAGE.to_string()),
+            reason: None,
+        })
+    }
+}
+
+impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Cybersource
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,
+{
+    fn build_headers(
+        &self,
+        req: &types::RouterData<Flow, Request, Response>,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let date = OffsetDateTime::now_utc();
+        let cybersource_req = self.get_request_body(req)?;
+        let auth: cybersource::CybersourceAuthType =
+            cybersource::CybersourceAuthType::try_from(&req.connector_auth_type)?;
```

Comment:
```suggestion
        let auth = cybersource::CybersourceAuthType::try_from(&req.connector_auth_type)?;
```

You can remove the explicit type declaration. Since you're using `CybersourceAuthType::try_from()`, the type of the variable is evident.